pkg/server/ca/manager.go

```diff
@@ -454,9 +454,11 @@ func (m *manager) loadKeypairSets(ctx context.Context) error {
 		m.next.Reset()
 	}
 
-	if m.current.x509CA != nil && m.next.x509CA != nil &&
-		m.current.x509CA.cert.NotBefore.After(m.next.x509CA.cert.NotBefore) {
-		// swap the current and next keypair to get ascending order
+	// if B (next) was loaded but A (current) was not, OR if B comes before
+	// A, then swap B into the current slot.
+	if (m.current.x509CA == nil && m.next.x509CA != nil) ||
+		(m.current.x509CA != nil && m.next.x509CA != nil &&
+			m.current.x509CA.cert.NotBefore.After(m.next.x509CA.cert.NotBefore)) {
```

Comment:
Does this mean we might switch to B prematurely, e.g. if B was prepared but not activated?

EDIT: see below test case comment

Reply:
hmm, that shouldn't happen. if B is prepared but not activated, then both A and B will exist and A will come before B.

Reply:
Ah yes ok... I misread the logic

not this thing or that thing and not before but after this other not before 😆